### DISCUSSION TITLE
Don't work exchanging messages between remote candidates.

### DISCUSSION THREAD

**Discussion by daaaanil81** [pion/webrtc#2010] [Q&A] [answered]
I create the same system for my university, but I found a problem with my system. The local and remote candidates received real IPs addresses from the STUN server, but when the Local candidates sent STUN requests to the remote candidates. The remote candidate doesn't receive STUN requests from local candidates, but in local network(192.168.0.13) system works without problems. I don't know the reason why I found this problem. Help me, please. I have a small time. I started to work with GoLang last month. I think the problem is with the UDP connection, Tell me, please. Maybe I open no correctly connection. 

```
		connectionUDP, err = net.ListenUDP("udp4", &net.UDPAddr{
			IP:   net.ParseIP("0.0.0.0"),
		})
```

**Comment by daaaanil81** [comment]
Could @Sean-Der help me, please?

**Comment by Sean-Der** [comment, marked as answer]
Hi @daaaanil81 

STUN will not work in all cases. You can have some networks where filtering + mapping is address dependent. See [WebRTC for the Curious](https://webrtcforthecurious.com/docs/03-connecting/#nat-mapping) to learn more about it.

To connect users in all cases you will need to run a TURN server. Or run one host not behind a NAT.
